### PR TITLE
Add CloudWatchEvent rule name and description fields

### LINKED ISSUE #2832: CloudWatchEvent rule name and description fields
```
functions:
  run:
    handler: handler.run
    events:
      - schedule:
          rate: rate(10 minutes)
          enabled: false
          name: "I want my own name prefix here"
          description: "I want my own description here"
```
When deploying lambda with scheduled events, a cloud watch event rule gets created with the schedule specified in yml. How can we add the name and description field? Currently, it adds an ugly long name with no description.

### PR DESCRIPTION
**Pull request by horike37**
## What did you implement:

Closes #2832

## How did you implement it:
It is possible to add name and description property of scheduled event in CloudFormation.
These are not required properties.

## How can we verify it:
set property as follow
```yml
 events:
      - schedule:
          name: my-scheduled-event
          description: 'my scheduled event'
          rate: rate(10 minutes)
```

deployed name and description to a scheduled event Lambda 

## Todos:
Linting errors occurs in master branch. But my implementaion is no problem.
- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Enable ["Allow edits from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) for this PR
- [x] Change ready for review message below

***Is this ready for review?:*** YES

### CODE REVIEW

**Comment by horike37** [comment]
👍👍👍

File: docs/providers/aws/events/schedule.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ functions:
     handler: statistics.handler
     events:
       - schedule:
+          name: your-scheduled-rate-event-name
```
Comment (mthenw) [inline docs/providers/aws/events/schedule.md]: I think it's better to leave existing examples as they are and add another one with name and description.

File: docs/providers/aws/events/schedule.md [hunk 2]
```diff
@@ -46,3 +46,15 @@ functions:
...
+## Specify Name and Description
+
+Name and Description can be specified for a schedule event. These are not required properties.
+
+```yaml
+events:
+  - schedule:
+      name: your-scheduled-rate-event-name
+      description: 'your scheduled rate event description'
+      rate(2 hours)
```
Comment (mthenw) [inline docs/providers/aws/events/schedule.md]: looks like typo :)

**Comment by horike37** [comment]
@mthenw 
Thanks!! fixed it.

**Pull request merged**